internal/format/format.go

```diff
@@ -47,8 +47,7 @@ const BytesPerLine = ColumnsPerLine / 4 * 3
 
 // NewlineWriter returns a Writer that writes to dst, inserting an LF character
 // every ColumnsPerLine bytes. It does not insert a newline neither at the
-// beginning nor at the end of the stream, but it ensures the last line is
-// shorter than ColumnsPerLine, which means it might be empty.
```

Comment:
This behaviour must not be removed. It was explicitly added (as a breaking change to the format) to ensure that standalone parsing of stanzas will always terminate, which was necessary for the plugin protocol.

Reply:
I'm aware that this is a breaking change, but I was not aware why it was built this way and it seemed unintuitive to me. Could you give me a link to some documentation on the plugin protocol? I can't find anything about plugins in the spec.

Reply:
If the last line of a base64 section (like the stanza body) can be a full line, then a streaming parser can't distinguish between the end of a stanza, and an incomplete stanza (due to blocking or errors on IO). This was a problem for the plugin protocol, which is a sequence of stanzas. Requiring the last line to be short removes this ambiguity.

Reply:
Thanks for your answer, but I'm afraid I still don't fully understand the use-case. Why not just wait for the next stanza to begin or end of the header (`--- `)?

Reply:
In the header format, you can indeed block header parsing on receipt of the MAC line. In the plugin protocol you can't do that, because you need bidirectional communication, so one direction will pause after a stanza while waiting on the other direction. It either required a non-stanza separator in the plugin protocol (so plugin IPC was no longer just a stream of stanzas), or removal of the ambiguity from parsing; after discussion we opted for the latter.

Reply:
Thanks for the explanation, now I can understand the decision a little better! Is there some place I could read up on this "plugin protocol"? Is it supposed to become part of age/rage/the spec? Is the discussion, that led to your decision publicly visible somewhere?

Although I don't yet know what the plugin protocol is, I assume it is not going to become a fundamental component of age. This assumption reinforces my opinion, that the file format of age should not be complicated in order to cater to some non-primary-use-cases.

Reply:
The plugin protocol draft is here: https://hackmd.io/@str4d/age-plugin-spec

I would say that it's going to be the last fundamental component of age, as I expect most other things (YubiKey support, for example) to be built on top of that abstraction.

In general, I think a self-terminating encoding is less complicated per-se, not more. The issue here is that I reused the same code that I used in the internal format for the armor, which honestly might be a mistake.